kafka/consumer_test.go

```diff
@@ -46,13 +51,38 @@ func TestConsumerAPIs(t *testing.T) {
 		t.Fatalf("%s", err)
 	}
 
+	closedConsumer, err := NewConsumer(&ConfigMap{
+		"group.id":                 "gotest",
+		"socket.timeout.ms":        10,
+		"session.timeout.ms":       10,
+		"enable.auto.offset.store": false, // permit StoreOffsets()
+	})
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	closedConsumer.Close()
+
+	var nilConsumer *Consumer = nil
+
 	t.Logf("Consumer %s", c)
+	t.Logf("Consumer %s", closedConsumer)
+	t.Logf("Consumer %s", nilConsumer)
 
 	err = c.Subscribe("gotest", nil)
 	if err != nil {
 		t.Errorf("Subscribe failed: %s", err)
 	}
 
+	err = closedConsumer.Subscribe("gotest", nil)
```

Comment:
Rather than repeating this everywhere, perhaps we can have a method like `testConsumerAPIsErr(consumer, expectedErr)` and from TestConsumerAPIs, we just call 
```
testConsumerAPIsWithErr(c, nil)
testConsumerAPIsWithErr(closedConsumer, stateErr)
.... and so on
```

I understand that for the case where err is nil (the "valid" consumer), the handling may be different. If that is so, the method testConsumerAPIsErr can at least be used for the nil and the closed consumer, because their handling should be the same

Reply:
I agree